src/Form/ServerForm.php

```diff
@@ -101,8 +123,42 @@ public function form(array $form, FormStateInterface $formStaet) {
       }, $this->schemaManager->getDefinitions()),
       '#default_value' => $server->get('schema'),
       '#description' => t('The schema to use with this server.'),
+      '#ajax' => [
+        'callback' => '::ajaxSchemaConfigurationForm',
+        'progress' => [
+          'type' => 'throbber',
+          'message' => $this->t('Updating schema configuration form.'),
+        ],
+      ],
     ];
 
+    $form['schema_configuration'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Schema configuration'),
+      '#tree' => TRUE,
+    ];
+
+    $schema = (!empty($formState->getUserInput()['schema'])) ? $formState->getUserInput()['schema'] : $server->get('schema');
+    if (!empty($schema)) {
+      $schema_configuration = (!empty($server->get('schema_configuration')) && !empty($server->get('schema_configuration')[$schema])) ? $server->get('schema_configuration')[$schema] : [];
```

Comment:
We are using camelCase for all the code where it's possible within this module. There are some places where we can't do that due to interface restrictions or reflection, but in this case we can.

Everywhere where  possible, please use camelCase for all variables and function parameters.